mu_repo/__init__.py

```diff
@@ -69,6 +69,28 @@ def CreateConfig(config_file='.mu_repo'):
 def CreateParams(args, config_file='.mu_repo'):
     return Params(CreateConfig(config_file), args, config_file)
 
+
+def SearchConfigFile(start_dir, recurse_limit=20):
+    """Search upwards in the file system starting in start_dir for a ".mu_repo" file, until the maximum limit or
+    the root of the file-system is reached.
+    """
+    start_dir = os.path.normpath(os.path.abspath(start_dir))
+    while True:
+        filename = os.path.join(start_dir, '.mu_repo')
+        if os.path.isfile(filename):
+            return filename
+
+        recurse_limit -= 1
+        if recurse_limit == 0:
+            return None
+
+        parent_dir = os.path.abspath(os.path.join(start_dir, '..'))
```

Comment:
Instead of doing abspath at each step, as you did in the beginning, you could do os.path.dirname(start_dir) -- the check for stopping would be the same (it returns the same dirname if it's on the root).